sm/tedge_agent/src/main.rs

```diff
@@ -16,10 +33,17 @@ async fn main() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
-fn initialise_logging() {
+fn initialise_logging(debug: bool) {
+    let log_level = if debug {
+        tracing::Level::TRACE
+    } else {
+        tracing::Level::INFO
+    };
+
     tracing_subscriber::fmt()
         .with_timer(tracing_subscriber::fmt::time::ChronoUtc::with_format(
             "%Y-%m-%dT%H:%M:%S%.3f%:z".into(),
         ))
+        .with_max_level(log_level)
         .init();
 }
```

Comment:
How about moving this function to `tedge_utils` (maybe `logging` module) crate such that we can have consistent logging and formatting (if we want to change date we only would have to change it one place).
Both `mapper` and `agent` use `tedge_utils` atm anyway so there won't be additional overhead with imports and in the future if we need to limit amount of `features` to be imported by other crates we can do something along the lines:
```rust
tedge_utils = { path = "../../common/tedge_utils", default-features = false, features = ["fs", "logging"] }
```
etc.

Reply:
That's a good point.

Reply:
See https://github.com/thin-edge/thin-edge.io/pull/427/commits/625744f5e2da4b3b96b889b7f1e2a19c3e7bfa72